zetaclient/chains/solana/signer/signer.go

```diff
@@ -170,26 +186,69 @@ func (signer *Signer) TryProcessOutbound(
 	// set relayer balance metrics
 	signer.SetRelayerBalanceMetrics(ctx)
 
-	// broadcast the signed tx to the Solana network with preflight check
-	txSig, err := signer.client.SendTransactionWithOpts(
-		ctx,
-		tx,
-		// Commitment "finalized" is too conservative for preflight check and
-		// it results in repeated broadcast attempts that only 1 will succeed.
-		// Commitment "processed" will simulate tx against more recent state
-		// thus fails faster once a tx is already broadcasted and processed by the cluster.
-		// This reduces the number of "failed" txs due to repeated broadcast attempts.
-		rpc.TransactionOpts{PreflightCommitment: rpc.CommitmentProcessed},
-	)
-	if err != nil {
-		logger.Error().
-			Err(err).
-			Msgf("TryProcessOutbound: broadcast error")
-		return
+	// broadcast the signed tx to the Solana network
+	signer.broadcastOutbound(ctx, tx, chainID, nonce, logger, zetacoreClient)
+}
+
+// broadcastOutbound sends the signed transaction to the Solana network
+func (signer *Signer) broadcastOutbound(
+	ctx context.Context,
+	tx *solana.Transaction,
+	chainID int64,
+	nonce uint64,
+	logger zerolog.Logger,
+	zetacoreClient interfaces.ZetacoreClient,
+) {
+	// prepare logger fields
+	lf := map[string]any{
+		logs.FieldMethod: "broadcastOutbound",
+		logs.FieldNonce:  nonce,
+		logs.FieldTx:     tx.Signatures[0].String(),
 	}
 
-	// report the outbound to the outbound tracker
-	signer.reportToOutboundTracker(ctx, zetacoreClient, chainID, nonce, txSig, logger)
+	// try broacasting tx with increasing backoff (1s, 2s, 4s, 8s, 16s, 32s, 64s)
+	// to tolerate tx nonce mismatch with PDA nonce or unknown RPC error
+	backOff := broadcastBackoff
+	for i := 0; i < broadcastRetries; i++ {
+		time.Sleep(backOff)
+
+		// PDA nonce may already be increased by other relayer, no need to retry
+		pdaInfo, err := signer.client.GetAccountInfo(ctx, signer.pda)
+		if err != nil {
+			logger.Error().Err(err).Fields(lf).Msgf("unable to get PDA account info")
+		} else {
+			pda := contracts.PdaInfo{}
+			err = borsh.Deserialize(&pda, pdaInfo.Bytes())
```

Comment:
Can we move this in the Solana package maybe? To not have cryptography details in this package